### ISSUE TITLE
When distributed,granfana diagrams only have users  

### ISSUE THREAD

**Issue by zy7y** [SvenskaSpel/locust-plugins#86]
![image](https://user-images.githubusercontent.com/67154681/162168258-02930cf3-0bd1-478e-86a4-dcb10ab37f59.png)

# master code
```python
from locust import FastHttpUser, task
import locust_plugins
# from common.load_shapes import StagesShape


class Dummy(FastHttpUser):

    @task
    def hello(self):
        pass
```
# worker code
```python
from locust import FastHttpUser
from locust import task


class Dummy(FastHttpUser):
    host = "xxxxxx"
    @task
    def hello(self):
        self.client.get("/posts")
```

# my shell
```shell
locust --timescale --headless -f master.py --pguser postgres --pgpassword password --master --expect-workers=1

locust -f worker.py --workder
```
# use docker-compose.yml
```yml
version: "3"

services:
  postgres:
    image: cyberw/locust-timescale:2
    networks:
      - timenet
    expose:
      - 5432
    ports:
      # remove the 127.0.0.1 to expose postgres to other machines (including load gen workers outside this machine)
      - 127.0.0.1:5432:5432
    environment:
      # change the password if you intend to expose postgres to other machines
      - POSTGRES_PASSWORD=password
      - TIMESCALEDB_TELEMETRY=off
    volumes:
      - postgres_data:/var/lib/postgresql/data

  grafana:
    image: cyberw/locust-grafana:2
    environment:
      # these settings are not particularly safe, dont go exposing Grafana externally without changing this.
      - GF_AUTH_DISABLE_LOGIN_FORM=true
      - GF_AUTH_ANONYMOUS_ENABLED=true
      - GF_AUTH_ANONYMOUS_ORG_ROLE=Admin
      - GF_SECURITY_ALLOW_EMBEDDING=true
      - GF_LOG_LEVEL=warn # reduce log spamming. Remove this if you need to debug grafana.
    ports:
      - 127.0.0.1:3000:3000
    networks:
      - timenet
    volumes:
      - grafana_data:/var/lib/grafana

networks:
  timenet: null

volumes:
  postgres_data: null
  grafana_data: null
```
I'm very sorry. Could you tell me what I should do? Thank you

**Comment by cyberw** [comment]
Add `--timescale` to your worker as well. Maybe I can make that parameter be passed from master to worker automatically, but that should solve it for now.

**Comment by zy7y** [comment]
Thank you very much for your reply

Yes, RPS is available, but the number of users is not. The new worker command is as follows
```shell
 locust --timescale -f worker.py  --worker
```
```python
# new worker code
from locust import FastHttpUser
from locust import task
import locust_plugins


class Dummy(FastHttpUser):
    host = "xxxxxx"
    @task
    def hello(self):
        self.client.get("/posts")
```
# result
![image](https://user-images.githubusercontent.com/67154681/162189716-d7b4ee9b-7b9f-4359-bc2e-6329248fb3f1.png)

**Comment by zy7y** [comment]
> Add `--timescale` to your worker as well. Maybe I can make that parameter be passed from master to worker automatically, but that should solve it for now.
> 
> 添加---- 你的员工的时间表。也许我可以让这个参数自动地从主程序传递给工作者，但是现在应该可以解决这个问题了。

Thank you very much for your reply

Yes, RPS is available, but the number of users is not. The new worker command is as follows
```shell
 locust --timescale -f worker.py  --worker
```
```python
# new worker code
from locust import FastHttpUser
from locust import task
import locust_plugins


class Dummy(FastHttpUser):
    host = "xxxxxx"
    @task
    def hello(self):
        self.client.get("/posts")
```
# result
![image](https://user-images.githubusercontent.com/67154681/162189716-d7b4ee9b-7b9f-4359-bc2e-6329248fb3f1.png)

**Comment by zy7y** [comment]
> Add `--timescale` to your worker as well. Maybe I can make that parameter be passed from master to worker automatically, but that should solve it for now.
There seems to be a difference between distributed and non-distributed users
# The distributed result have tps users
![image](https://user-images.githubusercontent.com/67154681/162191442-67f9d42b-d6ed-4ab2-94a4-ccfbb15f869c.png)

# distributed  result 
![image](https://user-images.githubusercontent.com/67154681/162191760-1be13056-a336-46ba-ac1c-d54bf246a419.png)

Is there a complete example？

**Comment by vuongphamaioz** [comment]
My issue has been resolved since I found a solution.

**Comment by vuongphamaioz** [comment]
Hi @cyberw,

I fixed 2 things:
- The first is I used `locust.events.request_success` and `locust.events.request_failure` which are deprecated. These EventHook does not trigger the method `on_request` of the class `Timescale` in Locust plugins. Solution: I used `locust.events.request` instead.
- The second is when I ran distributed mode on different machines. I set up TimescaleDB + Grafana on the same machine with the `master` node. So the args I start Locust is
```bash
locust --timescale --master --pghost=localhost
```
The arg `--pghost=localhost` was passed to worker nodes and overwritten the `--pghost=<master-ip>` argument of worker nodes. So the workers send data to `localhost:5432` of themselves.
Solution: I set argument `--pghost=<master-ip>` on the `master` node. Then it becomes:
```bash
locust --timescale --master --pghost=<master-ip>
```
